src/components/PostDetails.tsx

```diff
@@ -2,105 +1,96 @@
+import { FC, useEffect, useState } from 'react';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { CommentsList } from './CommentsList';
+
+import { deleteComment, getComments } from '../api/comments';
+
+interface Props {
+  selectedPost: Post;
+}
+
+export const PostDetails: FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [error, setError] = useState('');
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const handleDeleteComment = (id: Comment['id']) => {
+    deleteComment(id)
+      .then(() =>
+        setComments(currentComments =>
+          currentComments.filter(comment => comment.id !== id),
+        ),
+      )
+      .catch(() => 'Unable to delete post');
+  };
+
+  useEffect(() => {
+    setIsShownForm(false);
+    setError('');
+    setIsLoadingComments(true);
+
+    getComments(selectedPost.id)
+      .then(setComments)
+      .catch(() => setError('Something went wrong'))
```

Comment:
The error message in the catch block for `getComments` is generic. Consider providing more specific error information to help users understand what went wrong.